Fix SlackAction: change SlackAction "value" to "url"

dvehar opened the pull request:
Using "value" does not work (maybe it used to). [Example](https://api.slack.com/docs/messages/builder?msg=%7B%22text%22%3A%22%3C%40W1A2BC3DD%3E%20approved%20your%20travel%20request.%20Book%20any%20airline%20you%20like%20by%20continuing%20below.%22%2C%22channel%22%3A%22C061EG9SL%22%2C%22attachments%22%3A%5B%7B%22fallback%22%3A%22Book%20your%20flights%20at%20https%3A%2F%2Fflights.example.com%2Fbook%2Fr123456%22%2C%22actions%22%3A%5B%7B%22type%22%3A%22button%22%2C%22text%22%3A%22Book%20flights%20%F0%9F%9B%AB%22%2C%22value%22%3A%22https%3A%2F%2Fflights.example.com%2Fbook%2Fr123456%22%7D%5D%7D%5D%7D)

Using "url" does work. [Example](https://api.slack.com/docs/messages/builder?msg=%7B%22text%22%3A%22%3C%40W1A2BC3DD%3E%20approved%20your%20travel%20request.%20Book%20any%20airline%20you%20like%20by%20continuing%20below.%22%2C%22channel%22%3A%22C061EG9SL%22%2C%22attachments%22%3A%5B%7B%22fallback%22%3A%22Book%20your%20flights%20at%20https%3A%2F%2Fflights.example.com%2Fbook%2Fr123456%22%2C%22actions%22%3A%5B%7B%22type%22%3A%22button%22%2C%22text%22%3A%22Book%20flights%20%F0%9F%9B%AB%22%2C%22url%22%3A%22https%3A%2F%2Fflights.example.com%2Fbook%2Fr123456%22%7D%5D%7D%5D%7D)

dvehar committed:
Remove dead code

dvehar committed:
Change SlackAction "value" to "url"

Using "value" does not work (maybe it used to). [Example](https://api.slack.com/docs/messages/builder?msg=%7B%22text%22%3A%22%3C%40W1A2BC3DD%3E%20approved%20your%20travel%20request.%20Book%20any%20airline%20you%20like%20by%20continuing%20below.%22%2C%22channel%22%3A%22C061EG9SL%22%2C%22attachments%22%3A%5B%7B%22fallback%22%3A%22Book%20your%20flights%20at%20https%3A%2F%2Fflights.example.com%2Fbook%2Fr123456%22%2C%22actions%22%3A%5B%7B%22type%22%3A%22button%22%2C%22text%22%3A%22Book%20flights%20%F0%9F%9B%AB%22%2C%22value%22%3A%22https%3A%2F%2Fflights.example.com%2Fbook%2Fr123456%22%7D%5D%7D%5D%7D)

Using "url" does work. [Example](https://api.slack.com/docs/messages/builder?msg=%7B%22text%22%3A%22%3C%40W1A2BC3DD%3E%20approved%20your%20travel%20request.%20Book%20any%20airline%20you%20like%20by%20continuing%20below.%22%2C%22channel%22%3A%22C061EG9SL%22%2C%22attachments%22%3A%5B%7B%22fallback%22%3A%22Book%20your%20flights%20at%20https%3A%2F%2Fflights.example.com%2Fbook%2Fr123456%22%2C%22actions%22%3A%5B%7B%22type%22%3A%22button%22%2C%22text%22%3A%22Book%20flights%20%F0%9F%9B%AB%22%2C%22url%22%3A%22https%3A%2F%2Fflights.example.com%2Fbook%2Fr123456%22%7D%5D%7D%5D%7D)

dvehar commented:
@gpedro @galimru please review

rgalimov commented:
Hi @dvehar, the `value` is needed for [interactive message buttons](https://api.slack.com/docs/message-buttons) as response value to help determine which button was pressed. When you use `url` field it will create link button like described here https://api.slack.com/docs/message-attachments#link_buttons 
I think it would be better to add `url` field and leave other fields as is.

dvehar committed:
Refactor SlackAction to use an object oriented approach for each type of action

Creates the concrete SlackActions:
- BasicButtonSlackAction
- InteractiveButtonSlackAction
- InteractiveSelectSlackAction

Adds tests

dvehar commented:
@rgalimov I pushed some changes. I broke out the SlackAction functionality into specific subclasses (`BasicButtonSlackAction`, `InteractiveButtonSlackAction`, `InteractiveSelectSlackAction`) and added tests.

dvehar committed:
BasicButtonSlackAction should extend ButtonSlackAction

dvehar commented:
@rgalimov can you please review?

galimru commented:
@dvehar this looks fine

dvehar commented:
@gpedro please review / merge.

dvehar commented:
@gpedro just following up. Can this change be merged?

deepakputhraya commented:
@gpedro @galimru Can these changes be merged?

dvehar commented:
@gpedro @jaypatel512 @rgalimov are you able to merge?

rgalimov commented:
@dvehar Sorry, I don't have access to merge PR

dvehar committed:
make style public
